src/boot.c

```diff
@@ -690,10 +690,16 @@ static void write_volume_label(DOS_FS * fs, char *label)
 {
     time_t now = time(NULL);
     struct tm *mtime = localtime(&now);
+    char *source_date_epoch = getenv("SOURCE_DATE_EPOCH");
     off_t offset;
     int created;
     DIR_ENT de;
 
+    if (source_date_epoch) {
+	    now = strtoull(source_date_epoch, NULL, 10);
```

Comment:
Does not check for garbage nor correct data in string. Also usage of strtoull is dangerous as it does not signal underflow. strtoll should be used. See also: https://github.com/dosfstools/dosfstools/pull/92